### PR TITLE
Sync go.(mod/sum) with argo-cd 1.8 release

### PR DESCRIPTION
**Pull request by kvendingoldo**

### CODE REVIEW

**Comment by kvendingoldo** [comment]
@jgwest @OmerKahani @wtam2018 @xianlubird can you check what is wrong with the unit test for generators/cluster, please? I see that it's the wrong order but in the general result the same.

**Comment by kvendingoldo** [comment]
@jgwest @dgoodwin @mgoodness @OmerKahani @wtam2018 @xianlubird please, review

File: pkg/utils/createOrUpdate.go [hunk 1]
```diff
@@ -11,17 +11,25 @@ import (
...
 
 // override "sigs.k8s.io/controller-runtime" CreateOrUpdate function to add equity function for argov1alpha1.ApplicationDestination
 // argov1alpha1.ApplicationDestination has a private variable, so the default implementation fails to compare it
-func CreateOrUpdate(ctx context.Context, c client.Client, obj runtime.Object, f controllerutil.MutateFn) (controllerutil.OperationResult, error) {
-	key, err := client.ObjectKeyFromObject(obj)
-	if err != nil {
-		return controllerutil.OperationResultNone, err
+func CreateOrUpdate(ctx context.Context, c client.Client, obj client.Object, f controllerutil.MutateFn) (controllerutil.OperationResult, error) {
+	key := client.ObjectKeyFromObject(obj)
+	if err := c.Get(ctx, key, obj); err != nil {
```
Comment (wtam2018) [inline pkg/utils/createOrUpdate.go:22]: Did you copy this if-block by mistake?  It looks like an exact copy of the one below it.

Comment (jgwest) [reply pkg/utils/createOrUpdate.go:22]: I had the same question as @wtam2018 , the first and second if block appear to be the same, unless there is a subtle difference hiding in there...

**Review by jgwest — CHANGES_REQUESTED** [review]
Thanks @kvendingoldo, looks great! Check out our question re: `createOrUpdate.go`, and let us know if this was intentional...

File: pkg/utils/createOrUpdate.go (hunk 1 above)
Comment (kvendingoldo) [reply pkg/utils/createOrUpdate.go:22]: It's copy-paste issue

**Comment by kvendingoldo** [comment]
@wtam2018 @jgwest it's fixed, thank you for your attention :)

**Review by jgwest — APPROVED** [review]
LGTM!

**Pull request merged**